parsec/class/list.h

```diff
@@ -868,9 +868,10 @@ parsec_list_nolock_push_sorted( parsec_list_t* list,
         parsec_list_nolock_push_front(list, newel);
     } else {
         /* take the range of priorities and decide whether to iterate forward or backward */
-        int tail_val = COMPARISON_VAL(_TAIL(list), off);
-        int head_val = COMPARISON_VAL(_HEAD(list), off);
-        if (COMPARISON_VAL(newel, off) > ((head_val + tail_val) / 2)) {
+        uint64_t tail_val = COMPARISON_VAL(_TAIL(list), off);
+        uint64_t head_val = COMPARISON_VAL(_HEAD(list), off);
+        uint64_t comp_val = COMPARISON_VAL(newel, off);
```

Comment:
According to the use we do of this macro the `priority` is an `int` and can be both positive or negative. By converting it to an `uint64_t` you are not correctly computing the median point, so it is not worse than the truncation that resulted from the initial code.

Reply:
Fixed to use signed integer.